### DISCUSSION TITLE
Possibility to filter or sort the checks (Github Actions) in the Pull Request view

### DISCUSSION THREAD

**Discussion by mnojek** [community/community#48154] [Actions]
### Select Topic Area

Product Feedback

### Body

I am contributing to the projects, which have dozens or even hundreds of checks running per each created PR.
This is very hard to scroll through them and find the ones that I am interested in at the moment.

For example, I sometimes want to see which of them are still running, or which of the required checks failed.

The only information I have is this not-very-friendly piece of sentence like "30 successful, 36 skipped, 2 failing, 1 in progress, and 1 expected checks" which takes a lot of time to process by my brain. Also, I have this list of all the checks, but it's narrow and it's hard to find stuff there.

I would imagine some kind of filters above the list of checks to quickly show only those I am interested in, like:
- filter only Required checks
- filter only failed checks
- filter checks by name
- filter checks by executor
- ...

I would maybe also order the checks based on their current status, e.g. at the top show only currently running, then Failed, then Passed and at the end the Skipped ones. There are many ways how this component can be improved and it needs to be reviewed, designed and implemented.

This will save the time spent on investigation and searching for the specific check. It will also improve visibility for other team members, because everyone will be able to quickly see what's happening.

**Comment by jsoref** [comment]
I wonder if we could get someone to make a github overlay to do this, I'd actually be willing to use a site that gave me these features...

**Reply by bdkent** [reply]
I was so frustrated with this I wrote a little browser extension: https://github.com/bdkent/gh-pr-ext

**Comment by Thynix** [comment]
To make this more specific, as an example, we have a monorepo and therefore many checks. Filtering checks by success/failure/skipped/cancelled and whatnot would be a great usability improvement. We regularly encounter things like this in the commit check summary / PR check summary, and finding anything specific requires a lot of scrolling and reading.

![image](https://github.com/community/community/assets/199917/ef6e7684-3b10-4e08-909e-bcbdb3100686)
![image](https://github.com/community/community/assets/199917/8f5f5093-8960-4e6b-bcdb-56b81ee1e5d5)

**Comment by NickPPC** [comment]
I am having the same problem. Having to scroll the whole list to find the few failed checks out of ~100 is not great.
If the `successful`. `failing`, and `skipped` could be clickable to filter the list (or anything similar) that would save users a lot of time

**Comment by JakeShirley** [comment]
+1 to this.  We have 150+ different checks and seeing `165 successful, 1 failed` is the most frustrating part of my day when I scroll to find it.  Even just a simple "bubble up checks that need attention" design would be nice.

**Comment by aspicer** [comment]
+2 to this. Please consider.

An additional feature request would be automatically rerunning failed tests n times.

**Comment by tvalentyn** [comment]
There is a search bar in https://github.com/<>/actions that can be useful sometimes. 

However it appears that the query doesn't support multiple allowed values for the same param, or negations.

**Comment by jtruemper-sap** [comment]
We are having the same issue. Example attached.. 2 checks out of 146 failed. Finding them isn't fun at all. ;-)

<img width="633" height="71" alt="image" src="https://github.com/user-attachments/assets/e1a9ef9d-03df-4dda-836b-268325871c8c" />

**Reply by jsoref** [reply]
Does Group by status help your use case?
[![Review required/Group by status or No grouping](https://github.com/user-attachments/assets/1f8695b8-0600-4a6e-af2b-47de8fa9ce84)](https://github.com/signavio/react-mentions/pull/777#pullrequestreview-3359232065)
